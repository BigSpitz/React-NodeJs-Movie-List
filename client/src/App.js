import './App.css';

const App = () => {
  return <div>Movie List App</div>;
};

export default App;
