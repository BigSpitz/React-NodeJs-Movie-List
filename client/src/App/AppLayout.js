import React from 'react';
import { Container } from '@material-ui/core';
import styled from 'styled-components';

import Header from './components/Header';
import Footer from './components/Footer';

const StyledContainer = styled.div`
  min-height: calc(100vh - 168px);
  margin-top: 20px;
  margin-bottom: 20px;
`;

const AppLayout = ({ children }) => {
  return (
    <>
      <Header />
      <StyledContainer>
        <Container>{children}</Container>
      </StyledContainer>
      <Footer />
    </>
  );
};

export default AppLayout;
