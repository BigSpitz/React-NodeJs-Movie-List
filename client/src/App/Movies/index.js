import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import MoviesList from './MoviesList';
import MoviesForm from './MoviesForm';
import MoviesPagination from './MoviesPagination';
import Loader from './components/Loader';
import {
  getMoviesAsync,
  selectMovies,
  selectLoading,
  getGenresAsync,
  selectGenres,
} from './moviesSlice';

const initialFormFields = {
  title: '',
  genre: '',
  page: 1,
};

const Movies = () => {
  const dispatch = useDispatch();
  const [searchFields, setSearchFields] = useState(initialFormFields);
  const { movies, totalItems } = useSelector(selectMovies);
  const loading = useSelector(selectLoading);
  const genres = useSelector(selectGenres);

  const handlePageChange = (e, page) => {
    setSearchFields({ ...searchFields, page });
  };

  useEffect(() => {
    dispatch(getMoviesAsync(searchFields));
  }, [searchFields, dispatch]);

  useEffect(() => {
    dispatch(getGenresAsync());
  }, [dispatch]);

  return (
    <>
      <MoviesForm setSearchFields={setSearchFields} genres={genres} />
      <MoviesPagination
        totalItems={totalItems}
        handlePageChange={handlePageChange}
        page={searchFields.page}
      />
      {loading ? <Loader /> : <MoviesList movieList={movies} />}
    </>
  );
};

export default Movies;
