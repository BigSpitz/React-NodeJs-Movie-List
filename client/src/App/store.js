import { configureStore } from '@reduxjs/toolkit';
import moviesReducer from './Movies/moviesSlice';

export const store = configureStore({
  reducer: {
    movies: moviesReducer
  }
});
