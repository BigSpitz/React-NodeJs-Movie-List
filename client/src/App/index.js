import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect,
} from 'react-router-dom';
import { ToastContainer } from 'react-toastify';

import AppLayout from './AppLayout';
import Movies from './Movies';
import './App.css';

import 'react-toastify/dist/ReactToastify.css';

const App = () => {
  return (
    <Router basename={process.env.PUBLIC_URL}>
      <Switch>
        <AppLayout>
          <ToastContainer />
          <Route exact path="/">
            <Movies />
          </Route>
        </AppLayout>
        <Route path="*">
          <Redirect to={process.env.PUBLIC_URL} />
        </Route>
      </Switch>
    </Router>
  );
};

export default App;
