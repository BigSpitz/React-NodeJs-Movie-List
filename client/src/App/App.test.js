import App from './index';
import { render } from '@testing-library/react';
import wrapComponent from '../utils/reduxWrapper';
import '@testing-library/jest-dom/extend-expect';

test('loads and displays app', () => {
  const app = render(wrapComponent(App));
  expect(app).toMatchSnapshot();
});
