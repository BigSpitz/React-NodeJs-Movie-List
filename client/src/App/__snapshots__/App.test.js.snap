// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Movie List Exercise
          </p>
        </div>
      </header>
      <div
        class="sc-gtsrHT hCoFzW"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <div
            class="Toastify"
          />
          <form
            class="sc-hKFxyN ivxeYW"
          >
            <div
              class="MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
                id="genre"
              >
                Genre
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                style="min-width: 150px;"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="genre"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                  id="genre"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  name="genre"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="margin: 0px 20px; max-width: 500px; flex-grow: 1;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
                for="title"
                id="title-label"
              >
                Movie Title
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  id="title"
                  name="title"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              style="margin-top: 15px;"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Search
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
          <div
            class="sc-eCApnc eEwIWl"
          >
            <nav
              aria-label="pagination navigation"
              class="MuiPagination-root"
            >
              <ul
                class="MuiPagination-ul"
              >
                <li>
                  <button
                    aria-label="Go to previous page"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiPaginationItem-icon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                      />
                    </svg>
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Go to next page"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiPaginationItem-icon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </button>
                </li>
              </ul>
            </nav>
          </div>
          <div
            class="sc-jSFjdj iUonLy"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-bdnxRM fQSmuZ"
      >
        Copyright 2021
      </div>
       
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Movie List Exercise
        </p>
      </div>
    </header>
    <div
      class="sc-gtsrHT hCoFzW"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="Toastify"
        />
        <form
          class="sc-hKFxyN ivxeYW"
        >
          <div
            class="MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
              id="genre"
            >
              Genre
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              style="min-width: 150px;"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="genre"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="genre"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                name="genre"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root"
            style="margin: 0px 20px; max-width: 500px; flex-grow: 1;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
              for="title"
              id="title-label"
            >
              Movie Title
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                id="title"
                name="title"
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            style="margin-top: 15px;"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Search
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
        <div
          class="sc-eCApnc eEwIWl"
        >
          <nav
            aria-label="pagination navigation"
            class="MuiPagination-root"
          >
            <ul
              class="MuiPagination-ul"
            >
              <li>
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiPaginationItem-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiPaginationItem-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="sc-jSFjdj iUonLy"
        >
          <div
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-bdnxRM fQSmuZ"
    >
      Copyright 2021
    </div>
     
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
